src/components/product-switcher/index.tsx

```diff
@@ -197,7 +200,9 @@ const ProductSwitcher: React.FC = () => {
       <button
         aria-controls={OPTION_LIST_ID}
         aria-expanded={isOpen}
-        aria-labelledby={generateSwitcherOptionIdFromProduct(currentProduct)}
+        aria-labelledby={generateSwitcherOptionIdFromProduct(
+          currentProduct || firstProduct
+        )}
```

Comment:
I didn't catch this when I reviewed earlier, but using `firstProduct` here will actually break the screen reader presentation of this component on the `/` page.

Here's a screenshot of the HTML for reference:

![image](https://user-images.githubusercontent.com/43934258/148140609-a5fae580-85e1-42c5-ad22-784ae2c68d8d.png)

So what happens when the `ProductSwitcher` comes into focus on the home page, the screen reader says "Terraform, collapsed, button." This is fine when we're on the terraform page (mostly fine, I think we could improve it but it's a good start), but not helpful from the homepage.

Short term fix I'm proposing for this PR:

- update `aria-labelledby` to be conditionally set like this:
  ```
  aria-labelledby={currentProduct ?
    generateSwitcherOptionIdFromProduct(currentProduct) :
    undefined
  }
  ```
- add an `aria-label` prop, and set it conditionally as well:
  ```
  aria-label={currentProduct ?
    undefined :
    'Product switcher'
  }
  ```

I don't like this solution long term, but it does ensure we have the correct screen reader presentation for now!

Reply:
Ah this is a great catch - really appreciated 💯  Have updated with your short term fix, and agreed on exploring what else we could do longer term 👍